examples/regression/2-advanced-analysis/plot_timeseries_enbpi.py

```diff
@@ -0,0 +1,291 @@
+]
+y_train = demand_train.loc[X_train.index, "Demand"]
+X_test = demand_test.loc[:, features]
+y_test = demand_test["Demand"]
+
+model_params_fit_not_done = False
+if model_params_fit_not_done:
+    # CV parameter search
+    n_iter = 100
+    n_splits = 5
+    tscv = TimeSeriesSplit(n_splits=n_splits)
+    random_state = 59
+    rf_model = RandomForestRegressor(random_state=random_state)
+    rf_params = {"max_depth": randint(2, 30), "n_estimators": randint(10, 100)}
+    cv_obj = RandomizedSearchCV(
+        rf_model,
+        param_distributions=rf_params,
+        n_iter=n_iter,
+        cv=tscv,
+        scoring="neg_root_mean_squared_error",
+        random_state=random_state,
+        verbose=0,
+        n_jobs=-1,
+    )
+    cv_obj.fit(X_train, y_train)
+    model = cv_obj.best_estimator_
+else:
+    # Model: Random Forest previously optimized with a cross-validation
+    model = RandomForestRegressor(
+        max_depth=10, n_estimators=50, random_state=59)
+
+# Estimate prediction intervals on test set with best estimator
+alpha = 0.05
+cv_MapieTimeSeries = BlockBootstrap(
+    n_resamplings=100, length=48, overlapping=True, random_state=59
+)
+
+mapie_plus = MapieTimeSeriesRegressor(
+    model, method="plus", cv=cv_MapieTimeSeries, agg_function="mean", n_jobs=-1
+)
+mapie_enpbi = MapieTimeSeriesRegressor(
+    model, method="plus", cv=cv_MapieTimeSeries, agg_function="mean", n_jobs=-1
+)
+
+gap = 1
+
+print("EnbPI, with no partial_fit, width optimization")
+mapie_enpbi = mapie_enpbi.fit(X_train, y_train)
+y_pred_npfit_enbpi, y_pis_npfit_enbpi = mapie_enpbi.predict(
+    X_test, alpha=alpha, ensemble=True, beta_optimize=True
+)
+coverage_npfit_enbpi = regression_coverage_score(
+    y_test, y_pis_npfit_enbpi[:, 0, 0], y_pis_npfit_enbpi[:, 1, 0]
+)
+width_npfit_enbpi = (
+    y_pis_npfit_enbpi[:, 1, 0] - y_pis_npfit_enbpi[:, 0, 0]
+).mean()
+
+print("EnbPI with partial_fit, width optimization")
+mapie_enpbi = mapie_enpbi.fit(X_train, y_train)
+y_pred_pfit_enbpi = np.zeros(y_pred_npfit_enbpi.shape)
+y_pis_pfit_enbpi = np.zeros(y_pis_npfit_enbpi.shape)
+
+
+y_pred_pfit_enbpi[:gap], y_pis_pfit_enbpi[:gap, :, :] = mapie_enpbi.predict(
+    X_test.iloc[:gap, :], alpha=alpha, ensemble=True, beta_optimize=True
+)
+
+for step in range(gap, len(X_test), gap):
+    mapie_enpbi.partial_fit(
+        X_test.iloc[(step - gap):step, :],
+        y_test.iloc[(step - gap):step],
+    )
+    (
+        y_pred_pfit_enbpi[step:step + gap],
+        y_pis_pfit_enbpi[step:step + gap, :, :],
+    ) = mapie_enpbi.predict(
+        X_test.iloc[step:(step + gap), :],
+        alpha=alpha,
+        ensemble=True,
+        beta_optimize=True,
+    )
+coverage_pfit_enbpi = regression_coverage_score(
+    y_test, y_pis_pfit_enbpi[:, 0, 0], y_pis_pfit_enbpi[:, 1, 0]
+)
+width_pfit_enbpi = (
+    y_pis_pfit_enbpi[:, 1, 0] - y_pis_pfit_enbpi[:, 0, 0]
+).mean()
+
+print("Plus, with partial_fit, width optimization")
```

Comment:
To remove or use `MapieRegressor` instead ?

Reply:
Removed